aiida/tools/graph/age_rules.py

```diff
@@ -208,7 +207,7 @@ def _init_run(self, operational_set):
             for proj_tag, projectionlist in projections.items():
                 try:
                     self._querybuilder.add_projection(proj_tag, projectionlist)
-                except InputValidationError:
+                except ValueError:
```

Comment:
`add_projection` can also raise `TypeError`, right? (line 1180 of `aiida.orm.querybuilder` or whereabouts) Are you not catching that one because there is no point in adding the extra info?

Btw, shouldn't we `except ... as ...` and `raise ... from ...` here?

Reply:
No reason, you are right, should also catch the `TypeError`.

Reply:
Btw, since I think you wrote this code: why do you raise a `KeyError` here and why has the string an explicit newline?

Reply:
Yes, now I don't remember if I added this or it was originally like that on the base AGE branch; in any case I can't think of any good reason so I was probably just blundering 😅 

If you want to change it though, is there a way to re-raise the same kind of exception? (since you are catching both `ValueException` and `TypeException`, you don't know a-priori which one to re-raise...)